tensorflow/core/kernels/linalg/matrix_solve_op.cc

```diff
@@ -357,13 +364,261 @@ class MatrixSolveOpGpu : public AsyncOpKernel {
   bool adjoint_;
 };
 
+#elif TENSORFLOW_USE_ROCM
```

Comment:
we need to merge the ROCm and CUDA implementations into one.

There are approx 250 lines of code here, out which maybe 15 are different between the ROCm and CUDA implementations.  We should be able to keep just those differences within `#if GOOGLE_CUDA .... #elif TENSORFLOW_USE_ROCM` blocks and keep the rest the same.

Reply:
Thanks Deven, I'll get on that.